akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -104,19 +104,27 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
   private def snapshotFile(metadata: SnapshotMetadata, extension: String = ""): File =
     new File(snapshotDir, s"snapshot-${URLEncoder.encode(metadata.persistenceId, "UTF-8")}-${metadata.sequenceNr}-${metadata.timestamp}${extension}")
 
-  private def snapshotMetadata(persistenceId: String, criteria: SnapshotSelectionCriteria): immutable.Seq[SnapshotMetadata] =
-    snapshotDir.listFiles(new SnapshotFilenameFilter(persistenceId)).map(_.getName).collect {
+  private def snapshotMetadata(persistenceId: String, criteria: SnapshotSelectionCriteria): immutable.Seq[SnapshotMetadata] = {
+    val files = snapshotDir.listFiles(new SnapshotFilenameFilter(persistenceId))
+    if (files eq null) Nil // if the dir was removed
```

Comment:
This only defers NPE to [here](https://github.com/akka/akka/blob/wip-16737-NPE-LocalSnapshotStore-patriknw/akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala#L47) or [here](https://github.com/akka/akka/blob/wip-16737-NPE-LocalSnapshotStore-patriknw/akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala#L66). Am I missing something?